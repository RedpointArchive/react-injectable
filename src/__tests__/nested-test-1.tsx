import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Inject } from '../index';

const FooContext = React.createContext<string>("foo");
const BarContext = React.createContext<string>("bar");

interface ComponentProps {
  foo: string;
  bar: string;
  baz: string;
  moo: string;
}

const Component = Inject(
  {
    foo: FooContext,
    bar: BarContext,
  },
  class Component extends React.Component<ComponentProps, {}> {
    public render() {
      return (
        <>
          {this.props.foo}
          {this.props.bar}
          {this.props.baz}
          {this.props.moo}
        </>
      );
    }
  }
);

interface DocumentProps {
  foo: string;
  bar: string;
}

class Document extends React.Component<DocumentProps, {}> {
  public render() {
    return (
      <FooContext.Provider value={this.props.foo}>
        <BarContext.Provider value={this.props.bar}>
          <Component baz="baz" moo="moo" />
        </BarContext.Provider>
      </FooContext.Provider>
    );
  }
}

it('nested injected component renders correctly', () => {
  const renderer = TestRenderer.create(<Document foo="foo2" bar="bar2" />);

  expect(renderer.toJSON()).toMatchSnapshot();
});
