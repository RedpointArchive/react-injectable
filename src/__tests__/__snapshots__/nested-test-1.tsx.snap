// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nested injected component renders correctly 1`] = `
Array [
  "foo2",
  "bar2",
  "baz",
  "moo",
]
`;
